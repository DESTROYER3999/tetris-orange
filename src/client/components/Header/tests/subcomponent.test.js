import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import HeaderSub from "../subcomponent";
import socket from "../../../services/socket-api";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

const props = {
  roomName: "Room",
  playerName: "Player",
  onBackHome: jest.fn()
};

describe("<HeaderSub />", () => {
  test("HeaderSub rendering with inside a room", () => {
    const wrapper = shallow(<HeaderSub {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("HeaderSub rendering outside of a room", () => {
    props.roomName = "";
    const wrapper = shallow(<HeaderSub {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
