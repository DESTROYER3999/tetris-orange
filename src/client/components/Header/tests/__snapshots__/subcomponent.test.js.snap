// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderSub /> HeaderSub rendering outside of a room 1`] = `
<styled.div
  id="header"
>
  <styled.div
    id="left-item"
  />
  <styled.div
    id="center-item"
  >
    <styled.h1
      data-text="Tetris Orange"
      id="title"
    >
      Tetris Orange
    </styled.h1>
  </styled.div>
  <styled.div
    id="right-item"
  >
    Player
  </styled.div>
</styled.div>
`;

exports[`<HeaderSub /> HeaderSub rendering with inside a room 1`] = `
<styled.div
  id="header"
>
  <styled.div
    id="left-item"
  >
    <styled.button
      onClick={[MockFunction]}
    >
      Home
    </styled.button>
  </styled.div>
  <styled.div
    id="center-item"
  >
    <styled.h1
      data-text="Tetris Orange"
      id="title"
    >
      Tetris Orange
    </styled.h1>
  </styled.div>
  <styled.div
    id="right-item"
  >
    Player
  </styled.div>
</styled.div>
`;
